keystone_performance/load_test.py

```diff
@@ -44,9 +46,77 @@ def stopProducing(self):
         pass
 
 
+class UserInfo(object):
+    def __init__(self, args):
+        auth_url = '%s/v3/auth/tokens' % args.url
+
+        if args.user_domain_id:
+            user_domain_info = {'id': args.user_domain_id}
+        else:
+            user_domain_info = {'name': args.user_domain_name}
+        if args.project_id:
+            project_info = {'id': args.project_id}
+        else:
+            project_info = {'name': args.project_name}
+        if args.project_domain_id:
+            project_info['domain'] = {'id': args.project_domain_id}
+        else:
+            project_info['domain'] = {'name': args.project_domain_name}
+        auth_req_body = {
+            'auth': {
+                'identity': {
+                    'methods': ['password'],
+                    'password': {
+                        'user': {
+                            'name': args.username,
+                            'domain': user_domain_info,
+                            'password': args.password
+                        }
+                    }
+                },
+                'scope': {
+                    'project': project_info
+                }
+            }
+        }
+
+        headers = {'content-type': 'application/json'}
+        r = requests.post(auth_url, json=auth_req_body, headers=headers)
+        if r.status_code != 201:
+            raise Exception('Error: Failed to get keystone token for getting ' +
+                'user info. Status code: ' + str(r.status_code))
+        token = r.headers['x-subject-token']
+
+        domains_url = '%s/v3/domains?name=Default' % args.url
+        headers = {'X-Auth-Token': token}
+        r = requests.get(domains_url, headers=headers)
+        if r.status_code != 200:
+            raise Exception('Error: Failed to get domain list. Status code: ' +
+                str(r.status_code))
+        domain_id = r.json()['domains'][0]['id']
+
+        users_url = '%s/v3/users?domain_id=%s' % (args.url, domain_id)
+        r = requests.get(users_url, headers=headers)
+        if r.status_code != 200:
+            raise Exception('Error: Failed to get user list. Status code: ' +
+                str(r.status_code))
+        users = r.json()['users']
+
+        self._user_infos = []
+        for user in users:
+            if user['name'].startswith('load_user'):
```

Comment:
Rather than doing the filtering in our code, this program should be able to do a query like ?name__startswith=load_user&domain_id=%s

http://specs.openstack.org/openstack/keystone-specs/api/v3/identity-api-v3.html#list-entities-filtered-by-attribute